fe/src/components/Header/Header.tsx

```diff
@@ -22,25 +19,31 @@ export default function Header(): JSX.Element {
       <Flex h={16} alignItems="center" justifyContent="space-between">
         <HStack spacing={8} alignItems="center">
           <Drawerbar />
-          <Heading as="h4" size="md">
+          <Heading as={Link} size="md" to="/">
             TPT-Loane
           </Heading>
           <HStack as="nav" spacing={4} display={{ base: 'none', md: 'flex' }}>
-            {NAV_ITEMS.map(navItem => (
-              <RouteLink key={navItem.href} to={navItem.href}>
-                <Box
-                  px={2}
-                  py={1}
-                  rounded="md"
-                  _hover={{
-                    textDecoration: 'none',
-                    bg: useColorModeValue('gray.200', 'gray.700'),
-                  }}
-                >
-                  <Text fontSize="xl">{navItem.label}</Text>
-                </Box>
-              </RouteLink>
-            ))}
+            {NAV_ITEMS.map(navItem => {
+              const { href, iconName } = navItem;
+              const newIconName: IconName = iconName as IconName;
```

Comment:
I'm not too sure about having a function this fat here @Kasparsu can agree.
Maybe separate a bit so this much logic isn't int the return?

Reply:
Jup. Its big and clunky. Should I make the entire Hstack a new component like HeaderLeftSide. Or what do you think I should do?

Reply:
You cult make it a entire new component or just extract into a separate function